Sources/Foundation/NSError.swift

```diff
@@ -738,6 +741,18 @@ public extension CocoaError {
         return _nsUserInfo[NSUnderlyingErrorKey] as? Error
     }
 
+    /// Return a list of underlying errors, if any. It includes the values of both NSUnderlyingErrorKey and NSMultipleUnderlyingErrorsKey. If there are no underlying errors, returns an empty array.
+    var underlyingErrors: [Error] {
+        var errors = [Error]()
+        if let underlyingError = _nsUserInfo[NSUnderlyingErrorKey] as? Error {
+            errors.append(underlyingError)
+        }
+        if let underlyingErrors = _nsUserInfo[NSMultipleUnderlyingErrorsKey] as? [Error] {
+            errors.append(contentsOf: underlyingErrors)
+        }
```

Comment:
Is the ordering well specified?  If not, I think it might be a good idea to swap the order here.  That way we can initialize the array and extend with the single element:

```swift
var errors = (_nsUserInfo[NSMultipleUnderlyingErrors] as? [Error]) ?? []
if let error = _nsUserInfo[NSUnderlyingErrorKey] as? Error {
  errors.append(error)
}
return errors
```

Reply:
I couldn't find about that, so I followed the ordering on macOS for now.